src/metricscollector/config/config.go

```diff
@@ -54,12 +55,21 @@ var defaultCollectorConfig = CollectorConfig{
 	PollInterval:    DefaultPollInterval,
 }
 
+type DataPruneConfig struct {
+	CutoffDays int `yaml:"cutoff_days"`
+}
+
+var defaultDataPruneConig = DataPruneConfig{
```

Comment:
Just thinking aloud for clarity,

We no longer want to have any pruning code in respective packages i.e in metricscollector or data aggregator etc

The approach is to have 

- autoscaler/src/pruner/cmd/ pruner.go -> running the specific ifrit pruner processesas follows
 -  autoscaler/src/pruner/metriccollector_metric_pruner.go
 - autoscaler/src/pruner/aggregator_metric_pruner.go etc